### PR TITLE
Add tests for AvoBatcher class

### PR DESCRIPTION
**Pull request by ekrisane**

### CODE REVIEW

File: src/__tests__/Batcher.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+    const events = AvoInspector.avoStorage.getItem(AvoBatcher.cacheKey);
+
+    expect(events).not.toBeNull();
+
+    // FIXME: expects Object or null
+    // @ts-ignore
+    expect(events.length).toEqual(1);
+
+    // FIXME: expects Object or null
```
Comment (tpom6oh) [inline src/__tests__/Batcher.ts:55]: What is the problem here?

File: src/tests/Sessions.test.ts [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
 jest.mock("../AvoNetworkCallsHandler");
 
-const defaultOptions = {
-  apiKey: "api-key-xxx",
-  env: AvoInspectorEnv.Prod,
-  version: "0",
-};
-
-// TODO: Reuse in class implementation and test
-const sessionTimeMs = 5 * 60 * 1000;
```
Comment (tpom6oh) [inline src/tests/Sessions.test.ts:18]: You ended up having separate constants in tests and in the app, right? I think it makes sense to have them separate.

**Comment by tpom6oh** [comment]
Looking at the other pr, batcher tests fail for some reason on node 12?

File: src/tests/Sessions.test.ts (hunk 2 above)
Comment (ekrisane) [reply src/tests/Sessions.test.ts:18]: Yes, that's right, created `constants.ts` file that stores all the constants that are used only in the tests and currently these values are not shared between tests and classes. So will leave it like that

File: src/__tests__/Batcher.ts (hunk 1 above)
Comment (ekrisane) [reply src/__tests__/Batcher.ts:55]: It expect that `getItem()` method returns object or null, but here the events is returned as an array

**Comment by ekrisane** [comment]
Hm, yes, will check. But should not be related to node v12, as here the Batcher tests do not fail on Node v12

File: src/__tests__/Batcher.ts (hunk 1 above)
Comment (tpom6oh) [reply src/__tests__/Batcher.ts:55]: I think it is designed to be an array, so you can put multiple session/schema payloads under the key and can access them all.

That's probably another test case, put a mix of sessions and schema payloads and see if all can be accessed from the storage.

Comment (tpom6oh) [reply src/__tests__/Batcher.ts:55]: and also there is a limit on saved events, if you try to put more than 1000 events oldest events should be dropped and only 1000 latest events should be in the storage

Comment (ekrisane) [reply src/__tests__/Batcher.ts:55]: > I think it is designed to be an array, so you can put multiple session/schema payloads under the key and can access them all.
> 
> That's probably another test case, put a mix of sessions and schema payloads and see if all can be accessed from the storage.

`getItem()` implementation defines that returned value is an object or null, but events are stored in storage as an array, yes. Will look mere into this 👍  For sessions/events there was a test case, where multiple events are stored in storage and then retrieved.

Comment (ekrisane) [reply src/__tests__/Batcher.ts:55]: > and also there is a limit on saved events, if you try to put more than 1000 events oldest events should be dropped and only 1000 latest events should be in the storage

Missed this limitation, will add tests for this 👍

**Pull request merged**